app/views/admin/users/_form.html.erb

```diff
@@ -1,4 +1,4 @@
-<%= render_component "error", object: current_user %>
+<%= render_component "error", object: f.object %>
```

Comment:
This tightly couples this partial to its caller. Can we pass this in as a local scope variable when we call it instead?

Also, I'm not sure what this change has to do with the rest of the PR. Why is this change needed?

Reply:
Actually I think you may have it the other way around: using `current_user` in a partial gives the partial a dependency on state that is not passed to the partial by the caller. The problem was that (as a root user on an edit user page)  when you updated the values and pressed save the error box (this code) was pulling its errors from `current_user` and not `@user` (as is set up by the controller).

In this case the coupling is intentional: the `_form` partial renders the form and the `error` partial renders the errors at the top of a form page

Reply:
I guess I just don't like to see `f.object` without any declaration of `f`, but it's not a blocker, I guess. It now makes more sense to me why this is here now I understand that there are actually fixes in this PR, not just tests.

Reply:
Ah.I see now. Yeah `f` (in this context) is defined [here](https://github.com/geeksforsocialchange/PlaceCal/blob/4af2a5447501e7f386ad8c3032cb73007efe103f/app/views/admin/users/new.html.erb#L9) and [here](https://github.com/geeksforsocialchange/PlaceCal/blob/4af2a5447501e7f386ad8c3032cb73007efe103f/app/views/admin/users/edit.html.erb#L11).

No it is not a good variable name (i think short letter names are okay in a small local scope like a loop but not as an argument). But the problem with PC is that the list of refactorable parts is literally endless and some of them may be dependent on this weird behaviour so i tend to take a light touch approach to fixing things as it can lead to scope creep